File: src/main/java/me/zeroeightsix/kami/event/events/ChunkEvent.java
package me.zeroeightsix.kami.event.events;

import me.zeroeightsix.kami.event.KamiEvent;
import net.minecraft.client.network.packet.ChunkDataS2CPacket;
import net.minecraft.world.chunk.Chunk;

/**
 * @author 086
 */
public class ChunkEvent extends KamiEvent {
    private Chunk chunk;
    private ChunkDataS2CPacket packet;

    public ChunkEvent(Chunk chunk, ChunkDataS2CPacket packet) {
        this.chunk = chunk;
        this.packet = packet;
    }

    public Chunk getChunk() {
        return chunk;
    }

    public ChunkDataS2CPacket getPacket() {
        return packet;
    }
}

File: src/main/kotlin/me/zeroeightsix/kami/event/events/ChunkEvent.kt
package me.zeroeightsix.kami.event.events

import me.zeroeightsix.kami.event.KamiEvent
import net.minecraft.network.packet.s2c.play.ChunkDataS2CPacket
import net.minecraft.world.chunk.Chunk

/**
 * @author 086
 */
open class ChunkEvent private constructor(val chunk: Chunk?) : KamiEvent() {
    class Load(chunk: Chunk?, val packet: ChunkDataS2CPacket) : ChunkEvent(chunk)
    class Unload(chunk: Chunk?) : ChunkEvent(chunk)
}
